### PR TITLE
Feature/update idea class methods

### PR DESCRIPTION
**Pull request by sme93**
# Pull Request Template
## Description
Please include a summary of the change and which issue is fixed. Please also include relevant screenshots of your code being deployed. 
Fixes # Uses our Idea class and Idea class methods to update local storage, delete from storage and favorite idea cards.

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Refactoring
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,13 +1,6 @@
 //****Global Variables ****
 var newIdea;
-//each time page refreshes, need to check local storage to populate ideas array with anything that was stored in there
-var retrievedIdeas = JSON.parse(localStorage.getItem("ideas"));
-//ternary -- functions like an if statement. if (retrievedIdeas) is truthy and holds any value, then ideas is assigned to retrievedIdeas from local storage. if not, ideas is assigned to an empty array
```
Comment (sme93) [inline src/main.js:5]: I took this logic from Riley and moved it into the class

**Review by jon-schlandt — APPROVED** [review]
Wow, nice job on those methods! I also dig how you were able to refactor the lines at the top of "main.js" and simply include them in our "renderPage()" function. 

One question I have is in regards to the logic on lines 20 thru 22 and 34 thru 36 in "idea.js". It seems to me that there would never be a case in which "retrievedIdeas" is an empty array because we would never get a chance to run "deleteFromStorage()" or "updateIdea()" if a card did not exist on the page (and therefore also exist in local storage).

**Comment by sme93** [comment]
> One question I have is in regards to the logic on lines 20 thru 22 and 34 thru 36 in "idea.js". It seems to me that there would never be a case in which "retrievedIdeas" is an empty array because we would never get a chance to run "deleteFromStorage()" or "updateIdea()" if a card did not exist on the page (and therefore also exist in local storage).

Yeah, you're right - we don't technically need it in the delete and update methods. We can try taking them out if you want!

**Comment by RMartin0717** [comment]
Great job with the logic while moving what we had in main.js over to idea.js and keeping the functionality!

**Pull request merged**